src/llama-grammar.cpp

```diff
@@ -944,9 +935,11 @@ void llama_grammar_accept(
             if (!llama_grammar_is_end_of_sequence(pos)) {
                 new_stack.push_back(pos);
             }
-            llama_grammar_advance_stack_memo(rules, new_stack, stacks_new, stacks_cache);
+            llama_grammar_advance_stack_memo(grammar->rules, new_stack, stacks_new, grammar->stacks_cache);
```

Comment:
I really like the signature simplification that you made for `llama_grammar_accept` -- passing around the `grammar` structure alone (rather than all of the individual pieces of the structure) feels good. 

Should we do that to `llama_grammar_advance_stack` / `llama_grammar_advance_stack_memo` as well?

I.E. something more like:

```
static void llama_grammar_advance_stack_memo(
        struct llama_grammar * grammar,
        const llama_grammar_stack  & stack,
              llama_grammar_stacks & new_stacks) {
```

?

Reply:
We should give it a try. I think I briefly tried it again, but there was some issue. Will retake a look tomorrow and see if we can simplify further.